[DOCS][9.x]  Improve ESQL reference docs information architecture

leemthompo opened the pull request:
## [ 👀URL preview](https://docs-v3-preview.elastic.dev/elastic/elasticsearch/pull/127248/reference/query-languages/esql)

This should make the reference more digestible via the side nav and therefore more easily navigable:

- reorganized es|ql reference from flat list to logical hierarchy
- created three main sections: syntax reference , special fields, advanced operations
- renamed pages with more consistent and task-oriented titles
- aligned navigation titles with page content
- improved introductory text for each section
- used parallel phrasing for similar concepts
- clarified the relationship between reference docs and conceptual docs

## Screenshots

### Nav before
<img width="188" alt="before" src="https://github.com/user-attachments/assets/94459b3e-88c0-4201-ba72-cc4df5174cd0" />


### Nav after 

<img width="193" alt="Screenshot 2025-04-25 at 09 39 25" src="https://github.com/user-attachments/assets/a30a8bc8-16f5-464a-b3a6-f565d043cbf4" />

leemthompo committed:
[DOCS][9.0]  Improve ESQL reference docs IA

- reorganized es|ql reference documentation from flat list to logical hierarchy
- created three main sections: language fundamentals, special fields, advanced operations
- renamed pages with more consistent and task-oriented titles
- updated toc.yml to reflect new hierarchical structure
- aligned navigation titles with page content
- improved introductory text for each section
- used parallel phrasing for similar concepts
- clarified the relationship between reference docs and conceptual docs

elasticsearchmachine commented:
Pinging @elastic/es-docs (Team:Docs)

elasticsearchmachine commented:
Pinging @elastic/es-analytical-engine (Team:Analytics)

leemthompo committed:
Finetune naming of sections and ordering

leemthompo committed:
tighten up vague verbiage

leemthompo committed:
clean up list description

leemthompo committed:
nitty edits

leemthompo committed:
delete silly heading

leemthompo committed:
de-vague

alex-spies commented on docs/reference/query-languages/esql.md:
```diff
@@ -1,33 +1,23 @@
-
-
-
+{{esql}} is a SQL-like query language for querying and analyzing time series data in {{es}}. It is designed to be easy to use and understand, while also being powerful enough to handle complex queries.
```
ESQL is not really SQL-like (except for the rare inspiration we draw from good ol' SQL), and it's not limited to time series data. Maybe like this?
```suggestion
{{esql}} is a piped query language for exploring and analyzing data in {{es}}. It is designed to be easy to use and understand, while also being powerful enough to handle complex data processing.
```

alex-spies commented on docs/reference/query-languages/esql.md:
```diff
@@ -1,33 +1,23 @@
-
+{{esql}} is a SQL-like query language for querying and analyzing time series data in {{es}}. It is designed to be easy to use and understand, while also being powerful enough to handle complex queries.
 
+This reference section provides detailed technical information about {{esql}} features, syntax, and behaviors:
```
```suggestion
This reference section provides detailed technical information about {{esql}} features, syntax, and behavior:
```

alex-spies commented on docs/reference/query-languages/esql/esql-advanced-commands.md:
```diff
@@ -0,0 +1,7 @@
+
+You can perform complex data transformations and enrichment with {{esql}} with the following commands:
+
+* [Extract data with `DISSECT` and `GROK`](/reference/query-languages/esql/esql-process-data-with-dissect-grok.md): Learn how to extract and transform structured data from unstructured text using the `DISSECT` and `GROK` commands.
+* [Augment data with `ENRICH`](/reference/query-languages/esql/esql-enrich-data.md): Learn how to combine data from different indices using the `ENRICH` command.
+* [Join data with `LOOKUP JOIN`](/reference/query-languages/esql/esql-lookup-join.md): Learn how to join data from different indices using the `LOOKUP JOIN` command.
```
nit: the links already say "with `DISSECT` and `GROK`", which is later repeated in the description "... using the `DISSECT` and `GROK` commands".

Maybe it's nicer like this?
```suggestion
* [Extract data with `DISSECT` and `GROK`](/reference/query-languages/esql/esql-process-data-with-dissect-grok.md): Learn how to extract and transform structured data from unstructured text.
* [Augment data with `ENRICH`](/reference/query-languages/esql/esql-enrich-data.md): Learn how to combine data from different indices.
* [Join data with `LOOKUP JOIN`](/reference/query-languages/esql/esql-lookup-join.md): Learn how to join data from different indices.
```

alex-spies commented on docs/reference/query-languages/esql/esql-types-and-fields.md:
```diff
@@ -0,0 +1,15 @@
+
+* [Time spans](esql-time-spans.md): Learn how to work with time intervals in {{esql}}.
+
+* [Metadata fields](esql-metadata-fields.md): Learn about the metadata fields supported by {{esql}}, including `index` and `score`.
```
```suggestion
* [Metadata fields](esql-metadata-fields.md): Learn about the metadata fields supported by {{esql}}, including `_index` and `_score`.
```

alex-spies commented on docs/reference/query-languages/esql/esql-types-and-fields.md:
```diff
@@ -0,0 +1,15 @@
+navigation_title: "Types and fields"
+---
+
+# Types and fields in {{esql}}
```
Maybe like this?
```suggestion
# Types and special fields in {{esql}}
```
We actually mean metadata fields, no?

leemthompo replied:
and mv, yup just trying to keep the umbrella general enough so that future pages might also slot in here :)

alex-spies reviewed (approved):
Thanks @leemthompo , looks good! I only left some minor remarks.

leemthompo committed:
Apply suggestions from code review

Co-authored-by: Alexander Spies <alexander.spies@elastic.co>

leemthompo committed:
Rename advanced commands to advanced workflows and emphasize these are guides

This diff updates the terminology and organization of the ESQL documentation to better reflect content purpose:

. Improves description on main ESQL page to be clearer and more future-proof
. Updates heading to "Advanced workflows in {{esql}}"
. Refines introduction text to emphasize these are guides, not just command references

elasticsearchmachine commented:
## 💚 Backport successful
| Status | Branch | Result |
|:------:|:------:|:------:|
| ✅ |  [9.0](https://github.com/elastic/elasticsearch/pull/127380)  | [<img src="https://img.shields.io/github/pulls/detail/state/elastic/elasticsearch/127380">](https://github.com/elastic/elasticsearch/pull/127380) |
